gui/mountPrimaryWindow.py

```diff
@@ -250,31 +257,28 @@ def modifyPipeline(self,target):
             self.errorMessageBox(
                 f"Could not read the config file in {currentDir}. Consider going through the settings for selected target.",
                 "Could not read config file")
-            return False
+            return
         
         # Modify the components (pipeline) in the config file. 
         #   If the components are not present, there's a chance the configs are wrong. Notify user.
 
         components = configs.get('components')
         if components != None:
-            components[mountTargetComponent] = target
+            components[mountTargetComponent] = targetBucket
             configs['components'] = components
         else:
             self.errorMessageBox(
                 f"The components is missing in {currentDir}/config.yaml. Consider Going through the settings to create one.",
                 "Components in config missing")
-            return False
+            return
```

Comment:
Since there is not return value now, is there any risk of a failure in `modifyPipeline` causing issues?

Reply:
It acts the same way the code did before, it just notifies the user that something went wrong and couldn't modify the file. There is a chance that the user might forget the pipeline wasn't modified and would try to mount the bucket anyway. This points out the need to verify the integrity of the config file before mounting, but that functionality isn't in the main page anyway. I could add that to the backlog

Reply:
But before, `mountBucket` was gating the mount with the return value from this function.
Now if I try to mount with a config file that has an invalid `components` section, it will go ahead and try to mount.
I just tried mounting using the CLI, with components missing from the `components` section, and it just goes ahead and builds a pipeline that won't work, and give an I/O error when the OS tries to interact with the mount - a sort of silent failure on mount.
We could (and maybe should) validate the pipeline in the Cloudfuse Go code, but we don't.
I could go either way here - if it's going to break the code organization that you chose to put this validation back in, then you can leave it and make it a backlog item. But if it's easy to get it back, it would be nice to keep it.